### PR TITLE
encoding docs

### PR DESCRIPTION
**Pull request by kanitw**
@yuhanlu  I think we should use table generator for this page too.  Could you please take over this PR?

I split it into a bit more sections, which would fit better with your table generator.

Thanks

### CODE REVIEW

**Comment by YuhanLu** [comment]
Done

File: _includes/table.html [hunk 1]
```diff
@@ -29,8 +29,8 @@
       endcapture %}{%
       capture typeWithLink %}{%
         include checkLink.html check="type" channel=include.source param=type %}{%
-      endcapture
-    %}<tr>
+      endcapture%}
```
Comment (domoritz) [inline _includes/table.html]: add space before %

File: src/encoding.ts [hunk 2]
```diff
@@ -22,13 +22,15 @@ export interface Encoding {
   y?: PositionFieldDef | ValueDef<number>;
 
   /**
-   * X2 coordinates for ranged `bar`, `rule`, `area`
+   * X2 coordinates for ranged `bar`, `rule`, `area`.
    */
+  // TODO: Ham need to add default behavior
```
Comment (domoritz) [inline src/encoding.ts]: TODO

File: src/fielddef.ts [hunk 3]
```diff
@@ -35,6 +35,9 @@ export interface ConditionalValueDef<T> extends ValueDef<T> {
 export interface FieldDef {
   /**
    * Name of the field from which to pull a data value.
+   *
+   *  __Default value:__ `false`
```
Comment (kanitw) [inline src/fielddef.ts]: false?????

Since this is required, there is no default value. 

Add __Required.__ before "Name of ..." and remove default value.

**Pull request merged**